protocols/v2/roles-logic-sv2/src/job_dispatcher.rs

```diff
@@ -227,9 +227,300 @@ impl GroupChannelJobDispatcher {
+                .expect("Failed converting hex to bytes");
+        assert_eq!(
+            target.to_vec(),
+            actual_block_hash,
+            "Computed block hash does not equal the actaul block hash"
+        );
+    }
+
+    #[test]
+    fn test_group_channel_job_dispatcher() {
+        let mut jobs_creators = JobsCreators::new(BLOCK_REWARD, new_pub_key()).unwrap();
+        let group_channel_id = 1;
+        jobs_creators
+            .new_group_channel(group_channel_id, true)
+            .unwrap();
+        //Create a template
+        let mut template = template_from_gen(&mut Gen::new(255));
+        template.future_template = true;
+        let jobs = jobs_creators
+            .on_new_template(&mut template)
+            .expect("Failed to create new job");
+        // create extended mining job
+        let extended_mining_job = jobs.get(&1).unwrap();
+        // create GroupChannelJobDispatcher
+        let ids = Arc::new(Mutex::new(Id::new()));
+        let mut group_channel_dispatcher = GroupChannelJobDispatcher::new(ids);
+        // create standard channel
+        let target = Target::from(U256::try_from(utils::extranonce_gen()).unwrap());
+        let standard_channel_id = 2;
+        let extranonce = Extranonce::try_from(utils::extranonce_gen())
+            .expect("Failed to convert bytes to extranonce");
+        let standard_channel = StandardChannel {
+            channel_id: standard_channel_id,
+            group_id: group_channel_id,
+            target: target,
+            extranonce: extranonce.clone(),
+        };
+        // call target function (on_new_extended_mining_job)
+        let new_mining_job = group_channel_dispatcher
+            .on_new_extended_mining_job(extended_mining_job, &standard_channel)
+            .unwrap();
+
+        // on_new_extended_mining_job assertions
+        let (future_job_id, test_merkle_root) = assert_on_new_extended_mining_job(
+            &group_channel_dispatcher,
+            &new_mining_job,
+            &extended_mining_job,
+            extranonce.clone(),
+            standard_channel_id,
+        );
+        // on_new_prev_hash assertions
+        if extended_mining_job.future_job {
+            assert_on_new_prev_hash(
+                &mut group_channel_dispatcher,
+                standard_channel_id,
+                future_job_id,
+                test_merkle_root,
+            )
+        }
+        assert_on_submit_shares(
+            &group_channel_dispatcher,
+            standard_channel_id,
+            future_job_id,
+        );
+    }
+
+    fn assert_on_new_extended_mining_job(
+        group_channel_job_dispatcher: &GroupChannelJobDispatcher,
+        new_mining_job: &NewMiningJob,
+        extended_mining_job: &NewExtendedMiningJob,
+        extranonce: Extranonce,
+        standard_channel_id: u32,
+    ) -> (u32, Vec<u8>) {
+        // compute test merkle path
+        let new_root = merkle_root_from_path(
```

Comment:
Does the comment [here](https://github.com/stratum-mining/stratum/issues/317#issuecomment-1324156303) affect this test?

Reply:
it shouldnt because the length of the extranonce regarding #317 is controlled by a config file whereas the extranonce in the test is always generated with the correct 32 bytes length. The `merkle_root_from_path()` function only works if the extranonce is 32 bytes